File: 219. Contains Duplicate II/Solution.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        ind = {}
        for i,v in enumerate(nums):
            if v in ind:
                if i-ind[v]<=k:
                    return True
            ind[v] = i
        return False

File: 219. Contains Duplicate II/Solution.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int> ind;
        for(int i=0; i<nums.size(); i++){
            if(ind.count(nums[i])){
                if(abs(i-ind[nums[i]])<=k){
                    return true;
                }
            }
            ind[nums[i]] = i;
        }
        return false;
    }
};
